pontoon/base/models.py

```diff
@@ -377,6 +357,87 @@ def add_translation(self, string, locale, user):
 
         translation.save()
 
+    def get_translation(self, plural_form=None):
+        """Get fetched translation of a given entity."""
+        translations = self.fetched_translations
+
+        if translations:
+            if plural_form:
+                translations = [t for t in translations if t.plural_form == plural_form]
+
+            translation = sorted(translations, key=lambda k: (-k.approved, k.date))[0]
+
+            return {
+                'fuzzy': translation.fuzzy,
+                'string': translation.string,
+                'approved': translation.approved,
+                'pk': translation.pk
+            }
+
+        else:
+            return {
+                'fuzzy': False,
+                'string': u'',
+                'approved': False,
+                'pk': None
+            }
+
+    @classmethod
+    def serialize(self, project, locale, paths=None):
+        """Get project entities with locale translations."""
+
+        entities = self.objects.filter(resource__project=project, obsolete=False)
+
+        if paths:
+            entities = entities.filter(resource__path__in=paths)
+
+        entities = (entities
+                    .prefetch_related(
+                        Prefetch(
+                            'translation_set',
+                            queryset=Translation.objects.filter(locale=locale),
+                            to_attr='fetched_translations'
+                        )
+                    )
+                    .prefetch_related('resource')
```

Comment:
You should be able to pass this as an argument to the first `prefetch_related` call.